src/services/mongoose/users.ts

```diff
@@ -5,6 +5,8 @@ import {
   RepositoryError,
 } from "@domain/errors"
 import { User } from "@services/mongoose/schema"
+import { caseInsensitiveRegex } from "@core/utils"
+import { getUsernameRegex } from "@config/app"
```

Comment:
This function should probably belong in domain.
We need a `checkedToUsername` that executes this validation  before passing it around. When we have an instance of the `Username` type we should assume it has already been validated.